fix proto generation

mjd95 opened the pull request:
## Changes

Fix https://github.com/improbable-eng/thanos/issues/1131

adrien-f commented:
Hi :wave: !

Thanks for the changes, I'm trying to test them here on OSX and first, it won't work if I don't `go get -u github.com/gogo/protobuf/version` before. And then, like you said, `cp` from BSD won't work.

Could you maybe look into what the Prometheus team did for this similar issue here: https://github.com/prometheus/prometheus/blob/master/scripts/genproto.sh I think it's better to move that to our own script :smile: !

mjd95 commented:
Thanks @adrien-f, I agree.  I've updated our `genprotos.sh` to fetch the dependencies in the same way as the Prometheus one does and removed the hack from the `Makefile`.

domgreen commented on scripts/genproto.sh:
```diff
@@ -18,13 +18,17 @@ if ! [[ $(${PROTOC_BIN} --version) =~ "3.4.0" ]]; then
 THANOS_ROOT=$(pwd)
-PROM_PATH="${THANOS_ROOT}/pkg/store/storepb"
-GOGOPROTO_ROOT="${THANOS_ROOT}/vendor/github.com/gogo/protobuf"
+PROM_PATH="${THANOS_ROOT}/pkg/store/prompb"
```
nit - inline `THANOS_ROOT`?
